exercises/sgf-parsing/canonical-data.json

```diff
@@ -0,0 +1,161 @@
+{
+    "exercise": "sgf-parsing",
+    "version": "1.0.0",
+    "cases": [
+        {
+            "description": "empty input",
+            "property": "parse",
+            "input": {
+                "encoded": ""
+            },
+            "expected": {
+                "error": "tree missing"
+            }
+        },
+        {
+            "description": "tree with no nodes",
+            "property": "parse",
+            "input": {
+                "encoded": "()"
+            },
+            "expected": {
+                "error": "tree with no nodes"
+            }
+        },
+        {
+            "description": "node without tree",
+            "property": "parse",
+            "input": {
+                "encoded": ";"
+            },
+            "expected": {
+                "error": "tree missing"
+            }
+        },
+        {
+            "description": "node without properties",
+            "property": "parse",
+            "input": {
+                "encoded": "(;)"
+            },
+            "expected": {
+                "properties": {}
+            }
+        },
+        {
+            "description": "single node tree",
+            "property": "parse",
+            "input": {
+                "encoded": "(;A[B])"
+            },
+            "expected": {
+                "properties": {
+                    "A": ["B"]
+                },
+                "children": []
+            }
+        },
+        {
+            "description": "properties without delimiter",
+            "property": "parse",
+            "input": {
+                "encoded": "(;A)"
```

Comment:
I'd like to bring special attention to this test case. Some tracks that implemented this exercise (including one that I'm a maintainer on) have `(;a)` instead. However, I think `(;A)` is a better test case. Since this apparently is testing that properties without values are rejected, the property should be otherwise-valid. So `A` is better than `a` (which might be rejected due to not being uppercase).

(Note that I don't know SGF but from a quick reading of https://www.red-bean.com/sgf/sgf4.html it does appear that a value is required, so `(;A)` will be rejected)

Reply:
So I suggest other tracks with the exercise to change `(;a)` if you have it to `(;A)`.